config/config.go

```diff
@@ -64,3 +71,11 @@ func New() (*Configuration, error) {
 	}
 	return &c, nil
 }
+
+func (cfg *Configuration) GetCacheURL(uuid string) string {
+	var buffer bytes.Buffer
+	buffer.WriteString(cfg.CacheURL.Host)
```

Comment:
The [host](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Examples) is just something like `example.com` or `some.indexed.site.com`.

I don't think it should include `http/https`. If you want to configure that, it should probably be a separate `scheme` property.

Reply:
You're right, but adding `scheme` rules out the ability to use a protocol relative url for the cache. Is that okay?

Reply:
I'd think an empty scheme would be protocol-relative?

Reply:
Yep.. I'll add a check to not append `:` if the scheme is empty.

Reply:
You’ll also want to make sure to check for variations in scheme:

Ex:

‘’
‘http’
‘http://‘

How about if scheme is not empty you:

- lowercase scheme
- search for :// amd replace that with ‘’

Then you can safely add :// when constructing your URI if scheme not empty. You can also add other checks like scheme contains ‘http’ but prob overkill.

Reply:
yeah... you could do that too, if it'll help you.

I'm ambivalent about that sort of flexibility in general. By implementing "fault tolerant" HTML-parsing & rendering, IE managed to hold the entire internet back for about a decade.

This obviously won't have that big of an impact... but the principle is one we can learn from. The more flexible your input, the more logic needed for your code to interpret it. Since you don't know if any users rely on that behavior, you're committed to maintaining it forever. Breaking changes become harder as the project gains wider adoption.

So in general, there's a maintenance cost associated with accepting flexible input. ...but in this case, the cost for those suggestions is very small. So feel free to go for it if the benefits to you outweigh those costs.